test/Base_Tests/src/Semantic/Arrow_Interop_Spec.enso

```diff
@@ -20,14 +21,19 @@ add_specs suite_builder =
         group_builder.specify "should allow for creating a new fixed Array in Arrow" <|
             arrow_array_constructor = new_arrow
             array_10 = arrow_array_constructor.new 10
-            v = Vector.from_polyglot_array array_10
-            v.length . should_equal 10
-            v.at 1 . should_equal Nothing
-            v.at 10 . should_fail_with Index_Out_Of_Bounds
-            v.set 1 10
-            v.at 1 . should_equal 10
-            v.set 8 127
-            v.set 8 128 . should_fail_with Unsupported_Argument_Types
+            builder = Arrow_Array_Builder.from_arrow_array array_10
```

Comment:
For a while I am asking myself, if this is the right pattern to follow? Shouldn't the `Vector.new_builder` and other builders be changed to receive a callback lambda function to perform the building in a clearly defined scope? E.g.
```ruby
v = Vector.new_builder builder->
    builder.set 8 127
    builder.set 1 10
```
it looks the similar and _feels more functional_. Moreover such a _lambda based approach_ is going to work fine in the IDE (unlike the current _unclosed_ builder).

As far as I can say the _lambda_ approach supports all use-cases except _creating builder and leaving it open_ - however that shall be an anti-pattern anyway, I believe/hope.

Reply:
@JaroslavTulach I love that proposal! I was worried that the current Vector Builder is very IDE unfriendly and will behave in weird ways. Your suggestion makes it safe. I'm all in for making this refactor, although I imagine it would be a separate PR as it's independent of Arrow.

Reply:
I would prefer to make this change in a separate PR as this is not entirely related.